apps/trading/components/withdraw-container/withdraw-container.tsx

```diff
@@ -1,27 +1,426 @@
+  useVegaTransactionStore,
+} from '@vegaprotocol/web3';
+import {
+  useAssetDetailsDialogStore,
+  type AssetFieldsFragment,
+  type AssetERC20,
+} from '@vegaprotocol/assets';
+import { z } from 'zod';
+import { Controller, useForm, useWatch } from 'react-hook-form';
+import {
+  useDialogStore,
+  useVegaWallet,
+  useWallet,
+  useDisconnect as useVegaDisconnect,
+} from '@vegaprotocol/wallet-react';
+import {
+  useAccount,
+  useAccountEffect,
+  useDisconnect as useEvmDisconnect,
+  useReadContracts,
+} from 'wagmi';
+import {
+  FormGroup,
+  TradingInput as Input,
+  TradingButton,
+  truncateMiddle,
+  TradingInputError,
+  Intent,
+  TradingRichSelect,
+  TradingOption,
+} from '@vegaprotocol/ui-toolkit';
+import { ConnectKitButton } from 'connectkit';
+import { useEffect } from 'react';
+import {
+  addDecimalsFormatNumber,
+  removeDecimal,
+  toBigNum,
+} from '@vegaprotocol/utils';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { type Account, useAccounts } from '@vegaprotocol/accounts';
+import { AccountType } from '@vegaprotocol/types';
+import { EmblemByAsset } from '@vegaprotocol/emblem';
+import { BRIDGE_ABI } from '@vegaprotocol/smart-contracts';
 
-export const WithdrawContainer = ({ assetId }: { assetId?: string }) => {
+import { useT } from '../../lib/use-t';
+import {
+  FormSecondaryActionButton,
+  FormSecondaryActionWrapper,
+} from '../form-secondary-action';
+import { Thresholds } from './thresholds';
+
+type Configs = Array<EthereumConfig | EVMBridgeConfig>;
+
+export const WithdrawContainer = ({
+  initialAssetId,
+}: {
+  initialAssetId?: string;
+}) => {
   const { pubKey } = useVegaWallet();
-  const createTransaction = useVegaTransactionStore((state) => state.create);
+  const { data } = useAccounts(pubKey);
+
+  const { config } = useEthereumConfig();
+  const { configs } = useEVMBridgeConfigs();
+
+  if (!config) return null;
+  if (!configs?.length) return null;
+
+  const allConfigs = [config, ...configs];
+
+  const accounts = data?.filter(
+    (a) => a.type === AccountType.ACCOUNT_TYPE_GENERAL
+  );
+
+  const account = accounts?.find((a) => a.asset.id === initialAssetId);
+  const asset = account?.asset;
+
   return (
-    <WithdrawFormContainer
-      assetId={assetId}
-      partyId={pubKey ? pubKey : undefined}
-      submit={({ amount, asset, receiverAddress }) => {
-        createTransaction({
-          withdrawSubmission: {
-            amount,
-            asset,
-            ext: {
-              erc20: {
-                receiverAddress,
-              },
-            },
-          },
-        });
-      }}
+    <WithdrawForm
+      accounts={accounts || []}
+      initialAssetId={asset?.id || ''}
+      configs={allConfigs}
     />
   );
 };
+
+const withdrawSchema = z.object({
+  assetId: z.string().min(1, 'Required'),
+  fromPubKey: z.string().regex(/^[A-Fa-f0-9]{64}$/i, 'Connect Vega wallet'),
+  toAddress: z.string().min(1, 'Enter address or connect wallet'),
```

Comment:
fix (blocking): we should validate the to address is valid eth address if this can be user input